common/tools/eslint-plugin-azure-sdk/tests/rules/ts-doc-internal-private-member.ts

```diff
@@ -0,0 +1,179 @@
+
+                /**
+                * Index signature
+                */
+                private [s: string]: boolean | ((s: string) => boolean);
+
+                /**
+                 * Parameter Property
+                 */
+                private testMethod(private x: number, private y: number) {}
+
+                /**
+                 * Method Definition
+                 */
+                private get getter(): number { return 0 }
+
+                /**
+                 * Method Signature
+                 */
+                private method1(): any;
+
+              }`,
+      filename: "src/test.ts"
+    },
+    // all internal
+    {
+      code: `
+            /**
+              * Class documentation
+              */
+              class ExampleClass {
+                /**
+                 * Class Property
+                 * @internal
+                 */
+                x = 0;
+
+                /**
+                 * Property Signature
+                 * @internal
+                 */
+                y: number;
+
+                /**
+                * Index signature
+                * @internal
+                */
+                [s: string]: boolean | ((s: string) => boolean);
+
+                /**
+                 * Parameter Property
+                 * @internal
+                 */
+                testMethod(private x: number, private y: number) {}
+
+                /**
+                 * Method Definition
+                 * @internal
+                 */
+                get getter(): number { return 0 }
+
+                /**
+                 * Method Signature
+                 * @internal
+                 */
+                 method1(): any;
+              }`,
+      filename: "src/test.ts"
+    }
+  ],
+  invalid: [
+    {
+      code: `
+             /**
+              * Class documentation
+              */
+              class ExampleClass {
+                /**
+                 * Class Property
+                 * @internal
+                 */
+                private x = 0;
+
+                /**
+                 * Property Signature
+                 * @internal
+                 */
+                private y: number;
+
+                /**
+                 * Index signature
+                 * @internal
+                 */
+                private [s: string]: boolean | ((s: string) => boolean);
+
+                /**
+                 * Parameter Property
+                 * @internal
+                 */
+                private testMethod(private x: number, private y: number) {}
+
+                /**
+                 * Method Definition
+                 * @internal
+                 */
+                private get getter(): number { return 0 }
+
+                /**
+                 * Method Signature
+                 * @internal
+                 */
+                private method1(): any;
+
+              }`,
+      filename: "src/test.ts",
+      errors: [
+        {
```

Comment:
do these errors have other information (like locations)? It would be nice to verify that too if they do.

Reply:
By location information do you mean line numbers or more descriptive messages such as which private member caused the error? At the moment, I don't think any other information is included in the error message

Reply:
@jeremymeng I think the eslint reporter should take care of attaching source info to the diagnostics.

Reply:
@witemple-msft my question is more about the expected errors are all the same so in theory it would not cover the case of a bug that causes an error is not reported while another false positive is reported.